fusestore/store/src/engine/mem_engine.rs

```diff
@@ -108,6 +116,33 @@ impl MemEngine {
         Ok(table_id)
     }
 
+    pub fn drop_table(
+        &mut self,
+        db_name: &str,
+        tbl_name: &str,
+        if_exists: bool
+    ) -> Result<(), Status> {
+        let r = self.dbs.get_mut(db_name).map(|db| {
+            let name2id_removed = db.table_name_to_id.remove_entry(tbl_name);
+            let id_removed = name2id_removed
+                .as_ref()
+                .and_then(|(_, id)| db.tables.remove(&id));
+            (name2id_removed, id_removed)
+        });
```

Comment:
@dantengsky as we discussed the last time when we met, I supposed that in your design the database and tables are stored as a file in DFS, but not a metadata record.

Thus `engine/mem_engine.rs` will be removed finally. It does not reflect the architecture anymore.
The only reason it has been kept is to let the tests pass.

E.g., an `ActionHandler` should get rid of the `MemEngine` ASAP

Reply:
Yeah,  metadata will be stored as files in DFS. But before PR 634 is closed, mem_engine is still needed. 

> The only reason it has been kept is to let the tests pass.

This PR is also mainly for the convenience of stateless test.